src/ruvsarpur.py

```diff
@@ -937,6 +945,19 @@ def runMain():
     deinit() #Deinitialize the colorama library
     
 
+def downloadSubtitles(subtitles, video_filename):
+  for subtitle in subtitles:
+    subtitleFilename = "{0}.{1}.webvtt".format(
+        video_filename.split(".mp4")[0], subtitle['name'])
+    r = requests.get(
```

Comment:
Please utilize the helper function `__create_retry_session()` that correctly handles retries, see examples in the code